clutch/src/main.rs

```diff
@@ -66,6 +67,66 @@ impl ReplTrait<F> for ClutchState<F> {
                 Expression::Sym(s) => {
                     if let Some(name) = s.simple_keyword_name() {
                         match name.as_str() {
+                            "COMMIT" => {
+                                let (first, rest) = store.car_cdr(&rest)?;
+                                let (second, _) = store.car_cdr(&rest)?;
+
+                                let (expr, secret) = if rest.is_nil() {
+                                    // TODO: also support Commitment::from_ptr_with_hiding (randomized secret at runtime).
+                                    (first, F::zero())
+                                } else if let Expression::Num(n) = store.fetch(&second).unwrap() {
+                                    (first, n.into_scalar())
+                                } else {
+                                    println!("Secret must be a Num");
+                                    return Ok(());
```

Comment:
The `?` all over this function will return an error if the expression isn’t of the expected type already. What’s the meaning of returning Ok here?

Reply:
I changed the outer error-handling and removed this. Thanks for the nudge.